internal/injector/typed/typecheck.go

```diff
@@ -124,3 +96,31 @@ func SplitPackageAndName(fullName string) (pkgPath string, localName string) {
 	localName = fullName[lastDot+1:]
 	return pkgPath, localName
 }
+
+// validateTypeNameIsInterface checks if a successfully looked-up types.Object represents
+// a type name that resolves to an interface. It assumes obj is not nil.
+func validateTypeNameIsInterface(obj types.Object, fullName string, pkgPath string, typeName string) (*types.Interface, error) {
+	typeObj, ok := obj.(*types.TypeName)
+	if !ok {
+		// Provide context whether it was expected to be built-in or package-qualified.
+		if pkgPath == "" {
+			return nil, fmt.Errorf("object %q is not a type name but a %T", typeName, obj)
+		}
+		return nil, fmt.Errorf("object %s.%s is not a type name but a %T", pkgPath, typeName, obj)
+	}
+
+	typ := typeObj.Type()
+	if !types.IsInterface(typ) {
+		// Use the original full name in the error message for clarity.
+		return nil, fmt.Errorf("type %s is not an interface", fullName)
+	}
+
+	// Use .Underlying() to get the *types.Interface.
+	iface, ok := typ.Underlying().(*types.Interface)
+	if !ok {
```

Comment:
Since those two checks effectvely do the same thing... why not simplify them top just one? Or is `.Underlying()` risky to call if it's not an interface?

Reply:
Good catch. It's just an oversight or a troubleshooting artifact. I'll remove.